timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceImpl.java

```diff
@@ -257,24 +260,22 @@ public ListenableFuture<TimestampRange> getFreshTimestampsAsync(int timestampsTo
     }
 
     @Override
-    public ListenableFuture<TimestampLeaseResponse> acquireTimestampLease(
-            TimestampLeaseName timestampName, UUID requestId, int numFreshTimestamps) {
+    public ListenableFuture<TimestampLeaseResponses> acquireTimestampLease(
+            UUID requestId, Map<TimestampLeaseName, Integer> numFreshTimestamps) {
         long timestamp = timestampService.getFreshTimestamp();
 
         Leased<LockToken> leasedLock = lockService
-                .acquireTimestampLease(timestampName, requestId, timestamp)
+                .acquireTimestampLease(requestId, numFreshTimestamps.keySet(), timestamp)
                 .get();
-        long minLeased = lockService.getMinLeasedTimestamp(timestampName).orElse(timestamp);
 
-        TimestampRange timestampRange = timestampService.getFreshTimestamps(numFreshTimestamps);
-
-        ConjureTimestampRange freshTimestamps =
-                ConjureTimestampRange.of(timestampRange.getLowerBound(), timestampRange.size());
+        Map<TimestampLeaseName, TimestampLeaseResponse> responses = KeyedStream.stream(numFreshTimestamps)
```

Comment:
I think this makes sense, but eventually we probably want to grab the sum of all requested timestamps in a single go?

Reply:
Possibly

This looked more readable than the alternative

Reply:
yeah, definitely good for now.